File: airbyte-commons-server/src/test/java/io/airbyte/commons/server/handlers/DeploymentMetadataHandlerTest.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.server.handlers;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.ArgumentMatchers.anyInt;
import static org.mockito.ArgumentMatchers.anyString;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import io.airbyte.api.model.generated.DeploymentMetadataRead;
import io.airbyte.commons.version.AirbyteVersion;
import io.airbyte.config.Configs;
import io.micronaut.context.env.Environment;
import java.util.Set;
import java.util.UUID;
import org.jooq.DSLContext;
import org.jooq.Result;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

class DeploymentMetadataHandlerTest {

  @ParameterizedTest
  @ValueSource(strings = {Environment.KUBERNETES, Environment.TEST})
  void testRetrievingDeploymentMetadata(final String activeEnvironment) {
    final UUID deploymentId = UUID.randomUUID();
    final String version = "0.1.2";
    final AirbyteVersion airbyteVersion = new AirbyteVersion(version);
    final Configs.DeploymentMode deploymentMode = Configs.DeploymentMode.OSS;
    final DSLContext dslContext = mock(DSLContext.class);
    final Environment environment = mock(Environment.class);
    final Result<org.jooq.Record> result = mock(Result.class);
    final String expectedEnvironment =
        Environment.KUBERNETES.equals(activeEnvironment) ? Configs.WorkerEnvironment.KUBERNETES.name() : Configs.WorkerEnvironment.DOCKER.name();

    when(result.getValue(anyInt(), anyString())).thenReturn(deploymentId);
    when(dslContext.fetch(anyString())).thenReturn(result);
    when(environment.getActiveNames()).thenReturn(Set.of(activeEnvironment));

    final DeploymentMetadataHandler handler = new DeploymentMetadataHandler(airbyteVersion, deploymentMode, dslContext, environment);

    final DeploymentMetadataRead deploymentMetadataRead = handler.getDeploymentMetadata();

    assertEquals(deploymentId, deploymentMetadataRead.getId());
    assertEquals(expectedEnvironment, deploymentMetadataRead.getEnvironment());
    assertEquals(deploymentMode.name(), deploymentMetadataRead.getMode());
    assertEquals(version, deploymentMetadataRead.getVersion());
  }

}

File: airbyte-commons-server/src/test/java/io/airbyte/commons/server/handlers/DeploymentMetadataHandlerTest.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.server.handlers

import io.airbyte.commons.version.AirbyteVersion
import io.airbyte.config.Configs.AirbyteEdition
import io.micronaut.context.env.Environment
import org.jooq.DSLContext
import org.jooq.Record
import org.jooq.Result
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource
import org.mockito.ArgumentMatchers
import org.mockito.Mockito
import java.util.UUID

internal class DeploymentMetadataHandlerTest {
  @ParameterizedTest
  @ValueSource(strings = [Environment.TEST])
  fun testRetrievingDeploymentMetadata(activeEnvironment: String) {
    val deploymentId = UUID.randomUUID()
    val version = "0.1.2"
    val airbyteVersion = AirbyteVersion(version)
    val airbyteEdition = AirbyteEdition.COMMUNITY
    val dslContext = Mockito.mock<DSLContext>(DSLContext::class.java)
    val environment = Mockito.mock<Environment>(Environment::class.java)
    val result: Result<Record?> = Mockito.mock(Result::class.java) as Result<Record?>

    Mockito.`when`<Any?>(result.getValue(ArgumentMatchers.anyInt(), ArgumentMatchers.anyString())).thenReturn(deploymentId)
    Mockito.`when`<Result<Record?>?>(dslContext.fetch(ArgumentMatchers.anyString())).thenReturn(result)
    Mockito.`when`<MutableSet<String?>?>(environment.getActiveNames()).thenReturn(setOf(activeEnvironment).toMutableSet())

    val handler = DeploymentMetadataHandler(airbyteVersion, airbyteEdition, dslContext)

    val deploymentMetadataRead = handler.getDeploymentMetadata()

    Assertions.assertEquals(deploymentId, deploymentMetadataRead.getId())
    Assertions.assertEquals(airbyteEdition.name, deploymentMetadataRead.getMode())
    Assertions.assertEquals(version, deploymentMetadataRead.getVersion())
  }
}
